subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -234,13 +234,29 @@ static void commissioned_cb(struct bt_enocean_device *device)
 #endif
 	status_pub(srv->mod, NULL, BT_MESH_SILVAIR_ENOCEAN_STATUS_SET, srv->addr.a.val);
 }
+
+static void decommissioned_cb(struct bt_enocean_device *device)
+{
+	struct bt_mesh_silvair_enocean_srv *srv;
+
+	srv = model_resolve(&device->addr);
+	if (!srv) {
+		return;
+	}
+	bt_addr_le_copy(&srv->addr, BT_ADDR_LE_NONE);
```

Comment:
When calling decommission_device it will clear the srv->addr before decommissioned_cb is called that way this will return before Status Set message is published.